pkg/agent/agent.go

```diff
@@ -155,6 +156,9 @@ func (k *Klocksmith) Run() error {
 		glog.Warningf("error waiting for an ok-to-reboot: %v", err)
 	}
 
+	// stop watching the update status
+	watchUpdateStatusStop <- struct{}{}
```

Comment:
That is in addition to this line, right? I don't think that will send something along the channel to tell it to close the connection.